edxadapt/static/js/src/edxadapt.js

```diff
@@ -65,8 +65,8 @@ function EdxAdaptXBlock(runtime, element) {
         })
         .fail(function(jqXHR, textStatus) {
             if (jqXHR.status == 404) {
-                // Student not found in EdxAdapt. Let's register them
-                registerUser();
+                // Student not found in EdxAdapt. Let's register him
+                $.when(configureUser()).then(registerUser());
```

Comment:
Great idea and I like syntax.

One thing though. You're putting function call inside `$.when()`, which expects to receive deferreds. 

But `configureUser()` returns nothing. So it looks like you need to change it to return a list of deferreds to get correct execution order.

Here's what jQuery documentation says:
> If a single argument is passed to jQuery.when() and it is not a Deferred or a Promise, it will be treated as a resolved Deferred and any doneCallbacks attached will be executed immediately.